src/vivarium/framework/population.py

```diff
@@ -220,8 +220,9 @@ def _validate_no_missing_initializers(initializers: Sequence[Tuple]) -> None:
             required_columns.extend(required)
 
         if not set(required_columns) <= set(created_columns):
-            raise PopulationError(f"The initializers {initializers} could not be added.  "
-                                  "Check for missing dependencies in your components.")
+            missing_columns = set(required_columns).difference(set(created_columns))
```

Comment:
maybe make this missing_columns set and then use it in the check (e.g., if missing_columns) so you don't have to do the set comparison twice?